### PR TITLE
Added --configure-for-azure option for running in Azure cloud

### PR DESCRIPTION
**Pull request by petemoore**
This PR adds `--configure-for-azure` option to generic-worker to unblock Azure work.

### CODE REVIEW

File: azure.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	err = json.Unmarshal(attestedDoc, &doc)
+	if err != nil {
+		return fmt.Errorf("Could not unmarshal attested document %q into AttestedDocument struct - is it valid JSON? %v", string(attestedDoc), err)
+	}
+	if doc.Encoding != "pkcs7" {
+		return fmt.Errorf(`Attested document has unsupported encoding (%q) - generic-worker only supports "pkcs7"`, doc.Encoding)
+	}
+
+	// TODO: when Azure Provider is completed in Worker Manager, we'll be able
+	// to do the following, but that isn't available yet:
```
Comment (imbstack) [inline azure.go]: This is just because the schema isn't there yet to generate this from?

Comment (petemoore) [reply azure.go]: That's right.

**Comment by djmitche** [comment]
This is ridiculous.  Let's talk as a team, or just you and me and coop, and settle this.

**Comment by djmitche** [comment]
I scheduled a meeting on Monday.

**Comment by petemoore** [comment]
It is likely we won't need to use this option, and we can use taskcluster-worker-runner instead. But rather than leave this PR open, or delete the code which may be useful if we hit issues with taskcluster-worker-runner, I'll merge for now. We can always remove this code later if it isn't needed, but no point in deleting it before we have a working alternative.

**Pull request merged**